src/common/SurgeLogger.cpp

```diff
@@ -0,0 +1,76 @@
+#include "SurgeLogger.h"
+#include "spdlog/sinks/basic_file_sink.h"
+#include "spdlog/sinks/stdout_sinks.h"
+
+#include "spdlog/logger.h"
+#include "spdlog/async.h"
+#include <memory>
+#include <filesystem.h>
+
+/*
+** This file exists to have a single global static which is initialized at library open time
+** thereby making sure that the log is set up as we want when it runs. 
+**
+** Two parts. The "init" function below sets up the logger in the OS reasonable way.
+** For now it is stderr+~/Library/Logs on macos and stderr only on windows and linux.
+**
+** The second part is the static variable which is initialized with the function call
+*/
+
+namespace SurgeLogInternal
+{
+
+typedef std::shared_ptr< spdlog::logger > log_t;
+namespace fs = std::experimental::filesystem;
+
+log_t internalSurgeLogInit()
+{
```

Comment:
Still puzzling around this:

- Could it be just `Log`, the typedef I mean?
- I guess you could use `static` here.

Reply:
I could use static on the function declaration but would it make a difference? It’s a symbol which appears in no headers. I forget what static does on a standalone function in modern C++ anymore.

I made a comment on your review about Log vs log_t. STL standard for typedefs is really `typedef foo bar_t` and so I followed that.  I’d really rather keep it that way. Or I could also just kill the typedef. Originally I had a class which people used so SurgeLogger::log_t is the idiomatic name and when I stopped exporting that class in my dev I kept the standard-ish name

Reply:
On the static thing: internalSurgeLogInit is not a member function of any class. So it is behaves just like a static member function of a single function class.

Reply:
OK, right, sorry I missed that one but so you did. I like clean binaries even if no other function (e.g. nm output and such).